src/utils/featheryClient/index.ts

```diff
@@ -507,9 +507,11 @@ export default class FeatheryClient extends IntegrationClient {
       body: formData
     };
     // Here we can safely remove the listener because offlineRequestHandler has its own beforeunload
-    this._removeCustomFieldListener();
     return this.offlineRequestHandler.runOrSaveRequest(
-      () => this._fetch(url, options, true, true),
+      () => {
+        this._fetch(url, options, true, true);
+        this._removeCustomFieldListener();
```

Comment:
after the fetch, we can remove the before unload listener because it's certain that changes have been submitted